Can this also address:
```bash
txorphanage.cpp:281:21: error: loop variable 'child_iter' creates a copy from type '__gnu_debug::_Safe_iterator<std::_Rb_tree_iterator<std::pair<const transaction_identifier<false>, TxOrphanage::OrphanTx>>, std::map<transaction_identifier<false>, TxOrphanage::OrphanTx>> const' [-Werror,-Wrange-loop-construct]
  281 |     for (const auto child_iter : iters) {
      |                     ^
txorphanage.cpp:281:10: note: use reference type '__gnu_debug::_Safe_iterator<std::_Rb_tree_iterator<std::pair<const transaction_identifier<false>, TxOrphanage::OrphanTx>>, std::map<transaction_identifier<false>, TxOrphanage::OrphanTx>> const &' to prevent copying
  281 |     for (const auto child_iter : iters) {
      |          ^~~~~~~~~~~~~~~~~~~~~~~
      |                     &
txorphanage.cpp:313:21: error: loop variable 'child_iter' creates a copy from type '__gnu_debug::_Safe_iterator<std::_Rb_tree_iterator<std::pair<const transaction_identifier<false>, TxOrphanage::OrphanTx>>, std::map<transaction_identifier<false>, TxOrphanage::OrphanTx>> const' [-Werror,-Wrange-loop-construct]
  313 |     for (const auto child_iter : iters) {
      |                     ^
txorphanage.cpp:313:10: note: use reference type '__gnu_debug::_Safe_iterator<std::_Rb_tree_iterator<std::pair<const transaction_identifier<false>, TxOrphanage::OrphanTx>>, std::map<transaction_identifier<false>, TxOrphanage::OrphanTx>> const &' to prevent copying
  313 |     for (const auto child_iter : iters) {
      |          ^~~~~~~~~~~~~~~~~~~~~~~
      |                     &
```

--- a/src/txorphanage.cpp
+++ b/src/txorphanage.cpp
@@ -278,7 +278,7 @@ std::vector<CTransactionRef> TxOrphanage::GetChildrenFromSamePeer(const CTransac
     // Convert to a vector of CTransactionRef
     std::vector<CTransactionRef> children_found;
     children_found.reserve(iters.size());
-    for (const auto child_iter : iters) {
+    for (const auto& child_iter : iters) {
         children_found.emplace_back(child_iter->second.tx);
     }
     return children_found;
@@ -310,7 +310,7 @@ std::vector<std::pair<CTransactionRef, NodeId>> TxOrphanage::GetChildrenFromDiff
     // Convert iterators to pair<CTransactionRef, NodeId>
     std::vector<std::pair<CTransactionRef, NodeId>> children_found;
     children_found.reserve(iters.size());
-    for (const auto child_iter : iters) {
+    for (const auto& child_iter : iters) {
         children_found.emplace_back(child_iter->second.tx, child_iter->second.fromPeer);
     }
     return children_found;

